#! /bin/sh
## 20_useless_assignment_fix.dpatch by  <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: What sort of utter crap does upstream use as a compiler anyway? fix same
## DP: type of error we once had in JetDirect::NewChannel...

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad --exclude=CVS --exclude=.svn ./io/hpiod/udevice.cpp ./io/hpiod/udevice.cpp
--- ./io/hpiod/udevice.cpp	2005-08-06 02:02:34.000000000 -0300
+++ ./io/hpiod/udevice.cpp	2005-08-06 02:03:10.122358657 -0300
@@ -165,7 +165,7 @@
 Channel *UniUsbDevice::NewChannel(unsigned char sockid)
 {
    Channel *pC=NULL;
-   int i, mode;
+   int i;
 
    /* Only support one channel. */
    if (ChannelCnt >= 1)
@@ -181,7 +181,6 @@
          pC->SetSocketID(sockid);
          pChannel[i] = pC;
          ChannelCnt++;
-         ChannelMode = mode;
          break;
       }
    }     
